pvlib/iotools/panond.py

```diff
@@ -0,0 +1,167 @@
+"""
+Get .PAN or .OND file data into a nested dictionary.
+"""
+
+
+def _num_type(value):
+    """
+    Determine if a value is float, int or a string
+    """
+    if '.' in value:
+        try:  # Detect float
+            value_out = float(value)
+            return value_out
+
+        except ValueError:  # Otherwise leave as string
+            value_out = value
+            return value_out
+
+    else:
+
+        try:  # Detect int
+            value_out = int(value)
+            return value_out
+
+        except ValueError:  # Otherwise leave as string
+            value_out = value
+            return value_out
+
+
+def _element_type(element):
+    """
+    Determine if an element is a list then pass to _num_type()
+    """
+    if ',' in element:  # Detect a list.
+        # .pan/.ond don't use ',' to indicate 1000. If that changes,
+        # a new method of list detection needs to be found.
+        values = element.split(',')
+        element_out = []
+        for val in values:  # Determine datatype of each value
+            element_out.append(_num_type(val))
+
+        return element_out
+
+    else:
+        return _num_type(element)
+
+
+def parse_panond(fbuf):
+    """
```

Comment:
I would vote than we only have a ``read_panond`` function and eliminate the ``parse_panond``.

We can always create a ``parse_panond`` function in the future, but it's annoying to deprecate it.

I believe the reasoning for having the ``parse`` function is to avoid duplicate code when you have a ``get`` and ``read`` function that need similar data handling.

Reply:
Good point that no `get` makes `parse` less relevant.  Aside from supporting `get`, `parse` is also helpful when file contents might be already in-memory (retrieved from a database or cloud storage, e.g.) instead of a file on disk.  Is that helpful for PAN/OND files?  The use cases I can think of feel rather contrived.  I will make the parse function private.